tools/msg_alltoall.py

```diff
@@ -0,0 +1,112 @@
+import torch 
+import torch.distributed as dist
+import itertools
+import numpy as np
+
+def alltoall_cpu(rank, world_size, output_tensor_list, input_tensor_list):
+    """Each process scatters list of input tensors to all processes in a cluster
+    and return gathered list of tensors in output list. The tensors should have the same shape.
+    Parameters
+    ----------
+    rank : int
+        The rank of current worker
+    world_size : int
+        The size of the entire
+    output_tensor_list : List of tensor
+        The received tensors
+    input_tensor_list : List of tensor
+        The tensors to exchange
+    """
+    input_tensor_list = [tensor.to(torch.device('cpu')) for tensor in input_tensor_list]
+    for i in range(world_size):
+        dist.scatter(output_tensor_list[i], input_tensor_list if i == rank else [], src=i)
+
+def alltoallv_cpu(rank, world_size, output_tensor_list, input_tensor_list):
+    """Each process scatters list of input tensors to all processes in a cluster
+    and return gathered list of tensors in output list.
```

Comment:
the only difference between the two functions is that one requires to have the same shape and the other doesn't?